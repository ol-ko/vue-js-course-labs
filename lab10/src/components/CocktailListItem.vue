<template>
  <router-link :to="{ name: 'cocktail', params: { id: cocktail.id } }" :class="$style.cocktail">
    <img :src="cocktail.imageUrl" :class="$style.image"/>
    <h3>{{ cocktail.title }}</h3>
    <p>{{cocktail.description}}</p>
  </router-link>
</template>

<script>
export default {
  props: {
    cocktail: {
      type: Object,
      validator(value) {
        if (!value.title || !value.description || !value.id || !value.imageUrl) {
          return false;
        }
        return true;
      }
    }
  }
};
</script>

<style module>
.cocktail {
  text-decoration: none;
  color: inherit;
}
.image {
  width: 100%;
}
</style>
