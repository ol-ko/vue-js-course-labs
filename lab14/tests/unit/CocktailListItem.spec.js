import { shallowMount } from '@vue/test-utils'
import CocktailListItem from '@/components/CocktailListItem.vue'

const testCocktail = {
  id: "TestCocktailId",
  title: "Test cocktail title",
  description: "Test cocktail description",
  imageUrl: "https://test-cocktail-image.jpg"
};

describe('CocktailListItem.vue', () => {
  it('renders cocktail data', () => {
    const wrapper = shallowMount(CocktailListItem, {
      propsData: {
        cocktail: testCocktail
      },
      stubs: ['router-link']
    });
    expect(wrapper.text()).toContain(testCocktail.title);
    expect(wrapper.text()).toContain(testCocktail.description);
  });
});
