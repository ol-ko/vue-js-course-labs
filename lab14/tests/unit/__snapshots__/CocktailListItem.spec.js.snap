// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailListItem.vue matches the snapshot 1`] = `
<router-link-stub to="[object Object]" class="cocktail"><img src="https://test-cocktail-image.jpg" class="image">
  <h3>Test cocktail title</h3>
  <p>Test cocktail description</p>
</router-link-stub>
`;
